src/support/intervals.cpp

```diff
@@ -36,26 +37,24 @@ IntervalProcessor::filterOverlaps(std::vector<Interval>& intervals) {
 
   std::sort(
     intIntervals.begin(), intIntervals.end(), [](const auto& a, const auto& b) {
-      return a.first.start < b.first.end;
+      return a.first.start < b.first.start;
     });
 
   std::vector<int> result;
-  auto& firstInterval = intIntervals[0];
+  auto& formerInterval = intIntervals[0];
```

Comment:
Making this a reference means that when you do `formerInterval = latterInterval` below, it writes to the first element in `intIntervals`, which is a little odd. Intervals should be small enough that copying them is cheap, so let's just make this a non-reference. Alternatively, to avoid copying intervals, you could make this an index.

Reply:
done